Source/WebCore/page/SecurityOrigin.cpp

```diff
@@ -179,18 +160,36 @@ SecurityOrigin::SecurityOrigin(const SecurityOrigin* other)
 
 Ref<SecurityOrigin> SecurityOrigin::create(const URL& url)
 {
-    if (RefPtr<SecurityOrigin> cachedOrigin = getCachedOrigin(url))
-        return cachedOrigin.releaseNonNull();
+    if (url.protocolIsBlob())
+        return forBlobURL(url);
 
     if (SecurityOriginData::shouldTreatAsOpaqueOrigin(url))
         return adoptRef(*new SecurityOrigin);
 
-    if (shouldUseInnerURL(url))
-        return adoptRef(*new SecurityOrigin(extractInnerURL(url)));
-
     return adoptRef(*new SecurityOrigin(url));
 }
 
+inline bool isSafelistedBlobProtocol(const URL& url)
+{
+    return url.protocolIsInHTTPFamily()
+        || url.protocolIsFile()
+        || LegacySchemeRegistry::schemeIsHandledBySchemeHandler(url.protocol());
```

Comment:
This is process-global.  Could you add an assertion that it is not used in the network process or gpu process?  Otherwise we'll have scope issues like the one I fixed in 263652@main

Reply:
It would really help me if you could explain this more. As well as why this is not a problem for `SecurityOriginData::shouldTreatAsOpaqueOrigin`.

The asserts cause everything to fail so clearly something else is needed here.